sklearn/ensemble/_hist_gradient_boosting/tests/test_grower.py

```diff
@@ -567,3 +567,86 @@ def test_ohe_equivalence(min_samples_leaf, n_unique_categories, target):
         assert predictor.get_max_depth() < predictor_ohe.get_max_depth()
 
     np.testing.assert_allclose(preds, preds_ohe)
+
+
+def test_grower_interaction_constraints():
+    """Check that grower respects interaction constraints."""
+    n_features = 6
+    interaction_cst = [{0, 1}, {1, 2}, {3, 4, 5}]
+    n_samples = 10
+    n_bins = 6
+    root_feature_splits = []
+
+    def get_all_children(node):
+        res = []
+        if node.is_leaf:
+            return res
```

Comment:
According to codecov, this is not being used. I think we can remove the `n.is_leaf` in the `for` loop:

```python
def get_all_children(node):
    res = []
    if node.is_leaf:
        return res
    for n in [node.left_child, node.right_child]:
        res.append(n)
        res.extend(get_all_children(n))
    return res
```